chain/chain/src/store.rs

```diff
@@ -1815,116 +1835,211 @@ impl<'a> ChainStoreUpdate<'a> {
         }
     }
 
-    pub fn clear_old_data_on_height(&mut self, height: BlockHeight) -> Result<(), Error> {
+    pub fn update_tail(&mut self, height: BlockHeight) {
+        self.tail = Some(height);
+    }
+
+    // Clearing block data of `block_hash`, if on a fork.
+    // Clearing block data of `block_hash.prev`, if on the Canonical Chain.
+    pub fn clear_block_data(
+        &mut self,
+        trie: Arc<Trie>,
+        mut block_hash: CryptoHash,
+        is_fork: bool,
+    ) -> Result<(), Error> {
+        let header = self
+            .get_block_header(&block_hash)
+            .expect("block data is not expected to be already cleaned")
+            .clone();
+        // Broken GC prerequisites found
+        assert_ne!(header.inner_lite.height, self.get_genesis_height());
+
         let mut store_update = self.store().store_update();
+
+        // 1. Apply revert insertions or deletions from ColTrieChanges for Trie
+        if is_fork {
+            // If the block is on a fork, we delete the state that's the result of applying this block
+            self.store()
+                .get_ser(ColTrieChanges, block_hash.as_ref())?
+                .map(|trie_changes: TrieChanges| {
+                    trie_changes
+                        .revert_insertions_into(trie.clone(), &mut store_update)
+                        .map_err(|err| ErrorKind::Other(err.to_string()))
+                })
+                .unwrap_or_else(|| Ok(()))?;
```

Comment:
Since it is a const value, it is enough to use `unwrap_or(Ok(()))` or even `unwrap_or_default()`

Reply:
++